File: Sesion 6/Marco teorico/13PARA QUÉ USAR MIEMBROS ESTÁTICOS/13.py
class Contador:
    contador_total = 0

    def __init__(self):
        Contador.contador_total += 1
        self.id = Contador.contador_total

    @staticmethod
    def get_total_contadores():
        return Contador.contador_total

    def get_id(self):
        return self.id


c1 = Contador()
c2 = Contador()
c3 = Contador()

print("Total de contadores creados:", Contador.get_total_contadores())
print("ID de c1:", c1.get_id())
print("ID de c2:", c2.get_id())
print("ID de c3:", c3.get_id())

File: Sesion 6/Marco teorico/13PARA QUÉ USAR MIEMBROS ESTÁTICOS/13.java
class Contador {
    private static int contadorTotal = 0;
    private int id;

    Contador() {
        contadorTotal++;
        id = contadorTotal;
    }

    static int getTotalContadores() {
        return contadorTotal;
    }

    int getId() {
        return id;
    }
}

public class Main {
    public static void main(String[] args) {
        Contador c1 = new Contador();
        Contador c2 = new Contador();
        Contador c3 = new Contador();

        System.out.println("Total de contadores creados: " + Contador.getTotalContadores());
        System.out.println("ID de c1: " + c1.getId());
        System.out.println("ID de c2: " + c2.getId());
        System.out.println("ID de c3: " + c3.getId());
    }
}
